### PR TITLE
Python 3 support

### PR DESCRIPTION
**Pull request by kannibalox**
Notes:
* Some function names like fmt.to_utf8 don't make sense in a py3 context, but I wanted this conversion to be as naive as possible
* Looks like someone made a stab at 3.4, I only tested 3.5

### CODE REVIEW

File: src/tests/test_xmlrpc2scgi.py [hunk 1]
```diff
@@ -154,7 +156,10 @@ def test_send(self):
     def test_scgi_request(self):
         resp = xmlrpc2scgi.scgi_request(MockedTransport("foo"), "bar", "baz")
         bad = "Bad response %s" % resp
-        self.failUnless(resp.startswith('b"26:CONTENT_LENGTH'), bad)
+        if six.PY3:
```
Comment (pyroscope) [inline src/tests/test_xmlrpc2scgi.py]: Never check for PY3, because PY4. Either PY2 or `not PY2`.

Comment (kannibalox) [reply src/tests/test_xmlrpc2scgi.py]: makes sense, I can't avoid it in fmt though

**Comment by kannibalox** [comment]
is this PR still needed or should i close it?

**Comment by kannibalox** [comment]
I'll close this, let me know if you need anything else,

**Comment by pyroscope** [comment]
I cherry-piucked, with one outstanding commit in my workdir.

**Pull request closed without merging**